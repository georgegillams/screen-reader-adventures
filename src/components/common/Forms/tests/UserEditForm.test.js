import React from 'react';
import { render } from '@testing-library/react';

import { UserEditForm } from '..';

describe('<UserEditForm />', () => {
  it('Should render correctly', () => {
    const { container } = render(<UserEditForm user={{}} onDataChanged={() => null} onSubmit={() => null} />);

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with admin controls shown', () => {
    const { container } = render(
      <UserEditForm showAdminControls user={{}} onDataChanged={() => null} onSubmit={() => null} />
    );

    expect(container).toMatchSnapshot();
  });
});
