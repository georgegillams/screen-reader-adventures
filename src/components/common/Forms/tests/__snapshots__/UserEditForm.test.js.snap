// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserEditForm /> Should render correctly 1`] = `
<div>
  <div
    class=""
  >
    <label
      class="forms__component forms__component__label"
      for="uname_FORM_ID"
    >
      Display name
    </label>
    <div
      class="input__outer forms__component forms__component__text-box"
      placeholder="Display name"
    >
      <input
        aria-disabled="false"
        aria-invalid="false"
        class="input__inner"
        id="uname_FORM_ID"
        name="Display name"
        spellcheck="false"
        value=""
      />
      <svg
        class="icon input__icon input__icon--valid"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
        />
      </svg>
    </div>
    <label
      class="forms__component forms__component__label"
      for="name_FORM_ID"
    >
      Real name
    </label>
    <div
      class="input__outer forms__component forms__component__text-box"
      placeholder="Real name"
    >
      <input
        aria-disabled="false"
        aria-invalid="false"
        autofill="name"
        class="input__inner"
        id="name_FORM_ID"
        name="Real name"
        spellcheck="false"
        value=""
      />
      <svg
        class="icon input__icon input__icon--valid"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
        />
      </svg>
    </div>
    <label
      class="forms__component forms__component__label"
      for="email_FORM_ID"
    >
      Email
    </label>
    <div
      class="input__outer forms__component forms__component__text-box"
      placeholder="Email"
    >
      <input
        aria-disabled="false"
        aria-invalid="false"
        autofill="email"
        class="input__inner"
        id="email_FORM_ID"
        name="Email"
        spellcheck="false"
        value=""
      />
      <svg
        class="icon input__icon input__icon--valid"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
        />
      </svg>
    </div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy forms__component forms__component__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Update
      </span>
    </button>
  </div>
</div>
`;

exports[`<UserEditForm /> Should render correctly with admin controls shown 1`] = `
<div>
  <div
    class=""
  >
    <label
      class="forms__component forms__component__label"
      for="uname_FORM_ID"
    >
      Display name
    </label>
    <div
      class="input__outer forms__component forms__component__text-box"
      placeholder="Display name"
    >
      <input
        aria-disabled="false"
        aria-invalid="false"
        class="input__inner"
        id="uname_FORM_ID"
        name="Display name"
        spellcheck="false"
        value=""
      />
      <svg
        class="icon input__icon input__icon--valid"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
        />
      </svg>
    </div>
    <label
      class="forms__component forms__component__label"
      for="name_FORM_ID"
    >
      Real name
    </label>
    <div
      class="input__outer forms__component forms__component__text-box"
      placeholder="Real name"
    >
      <input
        aria-disabled="false"
        aria-invalid="false"
        autofill="name"
        class="input__inner"
        id="name_FORM_ID"
        name="Real name"
        spellcheck="false"
        value=""
      />
      <svg
        class="icon input__icon input__icon--valid"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
        />
      </svg>
    </div>
    <label
      class="forms__component forms__component__label"
      for="email_FORM_ID"
    >
      Email
    </label>
    <div
      class="input__outer forms__component forms__component__text-box"
      placeholder="Email"
    >
      <input
        aria-disabled="false"
        aria-invalid="false"
        autofill="email"
        class="input__inner"
        id="email_FORM_ID"
        name="Email"
        spellcheck="false"
        value=""
      />
      <svg
        class="icon input__icon input__icon--valid"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
        />
      </svg>
    </div>
    <label
      class="checkbox forms__component"
      id="admin_FORM_ID"
    >
      <div
        class="checkbox__input-wrapper"
      >
        <input
          aria-invalid="false"
          aria-label="Is admin"
          class="checkbox__input"
          name="Is admin"
          type="checkbox"
        />
        <svg
          class="icon checkbox__check"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
          />
        </svg>
      </div>
      <span
        aria-hidden="true"
        class="checkbox__label"
      >
        Is admin
         
      </span>
    </label>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy forms__component forms__component__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Update
      </span>
    </button>
  </div>
</div>
`;
